client/python/lib/test/test_tensors.py

```diff
@@ -104,3 +104,135 @@ def test_make_ndarray_invalid_no_type():
+    assert tensor_proto.tensor_content == np_values.tobytes()
+    assert tensor_proto.dtype == DataType.DT_FLOAT
+    assert tensor_proto.tensor_shape == _shape
+
+def test_make_tensor_proto_valid_no_dtype_provided():
+    values = [1.0, 2.0, 3.0]
+    tensor_proto = make_tensor_proto(shape=[3], values = values)
+    np_values = np.array(values)
+    assert tensor_proto.tensor_content == np_values.tobytes()
+    assert tensor_proto.dtype == DataType.DT_DOUBLE
+    assert tensor_proto.tensor_shape == TensorShapeProto(dim = [TensorShapeProto.Dim(size=3)])
+
+def test_make_tensor_proto_valid_no_shape_provided():
+    values = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
+    tensor_proto = make_tensor_proto(dtype = DataType.DT_FLOAT, values = values)
+    expected_shape = TensorShapeProto(dim = [TensorShapeProto.Dim(size=2), TensorShapeProto.Dim(size=3)])
+    np_values = np.array(values, np.float32)
+    assert tensor_proto.tensor_content == np_values.tobytes()
+    assert tensor_proto.tensor_shape == expected_shape
+    assert tensor_proto.dtype == DataType.DT_FLOAT
+
+def test_make_tensor_proto_invalid_dtype_provided():
+    values = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
+    with pytest.raises(TypeError) as exception_info:
+        make_tensor_proto(shape = None, dtype = DataType.DT_STRING, values = values)
+
+    exception = exception_info.value
+    assert str(exception) == "[1.0, 2.0, 3.0] has type list, but expected one of: bytes"
+
+def test_make_tensor_proto_invalid_dimsions():
+    values = [[1.0, 2.0], [1.0, 2.0, 3.0]]
+    with pytest.raises(ValueError) as exception_info:
+        make_tensor_proto(shape=[2,3], dtype = DataType.DT_FLOAT, values = values)
+
+    exception = exception_info.value
+    assert str(exception) == "shape provided does not match values array shape"
+
+def test_make_tensor_proto_invalid_shape_type():
+    values = 5.0
+    with pytest.raises(TypeError) as exception_info:
+        make_tensor_proto(shape=0, dtype = DataType.DT_FLOAT, values = values)
+
+    exception = exception_info.value
+    assert str(exception) == "shape type should be list, but is int"
+
+def test_make_tensor_proto_invalid_dimensions_no_shape_provided():
+    values = [[1,2,3], [4,5]]
+    with pytest.raises(ValueError) as exception_info:
+        make_tensor_proto(shape=None, dtype = DataType.DT_INT8, values = values)
+
+    exception = exception_info.value
+    assert str(exception) == "setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part."
+
+def test_make_tensor_proto_invalid_dtype_as_float():
+    values = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
+    with pytest.raises(TypeError) as exception_info:
+        make_tensor_proto(shape = None, dtype = int, values = values)
+
+    exception = exception_info.value
+    assert str(exception) == "<class 'int'> has type type, but expected one of: int, long"
```

Comment:
Should we check if dtype provided is at least type int, long? @dkalinowski

Reply:
What is expected behavior of the library? To follow numpy behavior?

Reply:
The exception we are getting is from TensorProto() constructor call. Earlier we are calling:
```
np_dtype = _as_numpy_dtype(dtype)
np_values = np.array(values, np_dtype)
```
_as_numpy_dtype(dtype) is our function and as if for now if we do not provide dtype=DataType.* the function returns np.uint32.
This cause np.array() function to work properly.
Maybe we should check if isinstance(dtype, int) in _as_numpy_dtype() and raise an exception if it is not?

Reply:
Can we bring this part under `with` scope?